impeller/toolkit/interop/impeller.h

```diff
@@ -625,6 +625,71 @@ ImpellerContextCreateOpenGLESNew(
     ImpellerProcAddressCallback IMPELLER_NONNULL gl_proc_address_callback,
     void* IMPELLER_NULLABLE gl_proc_address_callback_user_data);
 
+//------------------------------------------------------------------------------
+/// @brief      Only available in instrumented builds with IMPELLER_DEBUG
+///             enabled, setting GL call logging will log all OpenGL calls along
+///             with their arguments.
+///
+///             This becomes spammy and is only meant to be used as a debugging
+///             aid in environments where traditional graphics debuggers like
+///             RenderDoc are not available.
+///
+///             If the function name is NULL, the option applies to all OpenGL
+///             calls made by Impeller.
+///
+///             Only calls made by Impeller will be logged. Impeller doesn't
+///             know about OpenGL calls made by the embedder.
+///
+///             Expect logging in the form of:
+///
+///             ```
+///             glDepthMask(1)
+///             glViewport(0, 0, 2048, 1536)
+///             glDepthRangef(0, 1)
+///             glDisable(2884)
+///             glFrontFace(2304)
+///             ```
+///
+///             This function does nothing if the context is not an OpenGL
+///             context.
+///
+/// @param[in]  context               The context
+/// @param[in]  opengl_function_name  The opengl function name or NULL if the
+///                                   option must be set for all OpenGL methods.
+/// @param[in]  enable                If logging must be enabled.
+///
+IMPELLER_EXPORT void ImpellerContextSetDebugGLCallLogging(
+    ImpellerContext IMPELLER_NONNULL context,
+    const char* IMPELLER_NULLABLE opengl_function_name,
+    bool enable);
+
+//------------------------------------------------------------------------------
+/// @brief      Only available in instrumented builds with IMPELLER_DEBUG
+///             enabled, setting GL error checking will trap on all OpenGL error
+///             after OpenGL calls made by Impeller.
+///
+///             There is appreciable overhead to this form of debugging and it
+///             must only be used in environments where traditional graphics
+///             debuggers like RenderDoc are not available.
+///
+///             If the function name is NULL, the option applies to all OpenGL
+///             calls made by Impeller.
+///
+///             Only calls made by Impeller will be logged. Impeller doesn't
+///             know about OpenGL calls made by the embedder.
+///
+///             This function does nothing if the context is not an OpenGL
+///             context.
+///
+/// @param[in]  context               The context
+/// @param[in]  opengl_function_name  The opengl function name
+/// @param[in]  enable                The enable
+///
+IMPELLER_EXPORT void ImpellerContextSetDebugGLErrorChecking(
```

Comment:
How do you expect these functions to be used?

Reply:
Some instances from earlier this week:

* We were trying to figure out why rendering after a save layer was broken. We enabled debugging all calls to bind and unbind framebuffer to discover that the framebuffer handle was not being delivered to Impeller. And, when that was fixed, Impeller was dropping [the handle on the floor](https://github.com/flutter/engine/commit/d99a4381b2c2a17f696c47bde275664c7d75e503). So `ImpellerContextSetDebugGLErrorChecking(context, "glBindFrambuffer", true)`
* I've used call logging to discover that texture initialization was nuking the contents that were provided by Android to Impeller [without informing Impeller](https://github.com/flutter/engine/commit/d99a4381b2c2a17f696c47bde275664c7d75e503#diff-01ec69d5f42a999ece57e003a9b746cf961212aa4c1384e02968e3f1b8b93bc4R500) which then re-initialized the contents. This one was weird because RenderDoc doesn't understand the Android extensions.
* The framebuffer binding issue manifested as an issue in the form a GL error on the first clear of the next frame. Enabling glGetError debugging after all calls led us to the bind framebuffer call in the first place.

Just some uses over the past week.

Reply:
Sorry, I meant how technically would someone who is developing use this to turn on debugging =). Do you want them to manually insert an invocation of this function somewhere?  There are no callers to it in the PR.

Reply:
Yeah, they'd manually insert this during debugging using some sort of binding.

Reply:
I think that's going to be kind of hard to find/use.  I think it would be easier if we just had an preprocessor macro for it.
```
// impeller/renderer/backend/gles/proc_table_gles.cc
#ifndef IMPELLER_LOG_GL_CALLS
  SetDebugGLCallLogging(true);
#endif  // IMPELLER_LOG_GL_CALLS
// If you want to just log messages a la carte you can use
// SetDebugGLCallLogging(true, "glClear");
```

Preprocessor macros aren't perfect but, proc table is the most logical place for someone to look for this.  This way someone can get the behavior they want by just throwing in a `#define IMPELLER_LOG_GL_CALLS` without mucking around trying to find out where to put a call and how to call it with the right arguments.

Reply:
> I think that's going to be kind of hard to find/use.

This was for standalone Impeller embedders. The "engine" is usually a prebuilt and the embedder is smaller and can interactively enable this with bindings.

Reply:
Ah, gotcha.

Reply:
Your note that there are no usages of this is valid though. Let me add an interop test too.